### PR TITLE
feat(product): add dependency injectable meta-reducers

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the new behavior?
- The product feature store config is passed by a token
- product state exposes a way to provide meta-reducers through DI

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
This is in preparation for #1642

### CODE REVIEW

File: libs/product/state/src/reducers/injection-tokens/public_api.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export {
+  daffProductProvideMetaReducers,
+  DAFF_PRODUCT_META_REDUCERS,
+} from './meta.token';
+export { DAFF_PRODUCT_STORE_CONFIG } from './config.token';
```
Comment (griest024) [inline libs/product/state/src/reducers/injection-tokens/public_api.ts:5]: remove from public api

Comment (griest024) [reply libs/product/state/src/reducers/injection-tokens/public_api.ts]: @damienwebdev removed

**Comment by lderrickable** [comment]
How will this help with https://github.com/graycoreio/daffodil/issues/1642 ?

**Comment by griest024** [comment]
> How will this help with #1642 ?

[I posted a flowchart](https://github.com/graycoreio/daffodil/issues/1642#issuecomment-876690766) but basically it will strip out related products from the load success action payload in order to dedupe them.

**Review by lderrickable — APPROVED** [review]

**Pull request merged**